### DISCUSSION TITLE
install on glassfish6 fails

### DISCUSSION THREAD

**Discussion by gerald2545** [eclipse-birt/birt#1337] [Q&A]
Hi, 
I tried to install birt 4.13.0 on glassfish6 (openjdk 11, on rocky linux 8)  - I copied birt.war in autodeploy folder of glassfish) but it fails with log : 

> [2023-06-14T17:36:52.410+0200] [glassfish 6.2] [SEVERE] [NCLS-CORE-00026] [jakarta.enterprise.system.core] [tid: _ThreadID=66 _ThreadName=AutoDeployer] [timeMillis: 1686757012410] [levelValue: 1000] [[
>   Exception during lifecycle processing
> java.lang.Exception: java.lang.IllegalStateException: ContainerBase.addChild: start: org.apache.catalina.LifecycleException: java.lang.IllegalArgumentException: java.lang.IllegalArgumentException: Unable to add listener of type: org.eclipse.birt.report.listener.ViewerServletContextListener, because it does not implement any of the required ServletContextListener, ServletContextAttributeListener, ServletRequestListener, ServletRequestAttributeListener, HttpSessionListener, or HttpSessionAttributeListener interfaces
>         at com.sun.enterprise.web.WebApplication.start(WebApplication.java:111)

Someone already succeeded in installing it on glassfish6?
I used this method to install birt 4.9 (If I remember correctly) on glassfish 5 without problem

Thank you for your help

Gérald

**Comment by claesrosell** [comment]
Glassfish 6 is a Jakarta EE9 implementation. This means that it does not use the javax.* -namespace but the Jakarta.* -namespace instead.
Since Birt currently is using the javax.* -namespace it is not compatible with Glassfish 6.

**Reply by gerald2545** [reply]
thank you for your quick and valuable answer..... 
I think birt 4.13 requires a recent version of java.....which is not compatible with glassfish 5.
I didn't find a compatibility table java/birt....does it exist?

Thank you again

**Reply by gerald2545** [reply]
i just checked, I currently run birt 4.8 on glassfish5 with jdk 1.8.0_372

**Reply by merks** [reply]
I'm not sure anything suggests that Glassfish 5 cannot run in Java 11 nor Java 17.  I only see mention that it uses JEE 8 and BIRT too is using JEE 8 components...

**Reply by gerald2545** [reply]
running flassfish 5.1 on java 11 leads to nullPointerException, and it seems to be known : https://github.com/eclipse-ee4j/glassfish/issues/22130
Is there a compatibility table between birt and application servers somewhere?
